jvm-streaming/src/Language/Java/Streaming.hs

```diff
@@ -48,37 +48,37 @@ isPoppableStream ref = do
         writeIORef ref (Streaming.cons x stream)
         return $ fromIntegral $ fromEnum True
 
-popStream :: Reflect a ty => IORef (Stream (Of a) IO ()) -> IO (J ty)
+popStream :: Reflect a ty => IORef (Stream (Of a) IO ()) -> IO (Ptr (J ty))
 popStream ref = do
     stream <- readIORef ref
     Streaming.uncons stream >>= \case
       Nothing -> do
         exc :: J ('Class "java.util.NoSuchElementException") <- new []
         JNI.throw exc
-        return jnull
+        return nullPtr
       Just (x, stream') -> do
         writeIORef ref stream'
-        reflect x
+        reflect x >>= JNI.getLocalRefPtr
```

Comment:
`reflect` should do the reverse of `reify`. Where `reify` transfers ownership of a value to Haskell, `reflect` should disclaim any ownership. So deleting any global ref sounds like the right thing to do. It's also a safe thing to do: Java will only get rid of the object if it itself doesn't have any reference to it. This is something we should do in `reflect`, not on a case-by-case basis.